Plugins/SpatialGDK/SpatialGDKEditorToolbar/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -456,6 +463,30 @@ void FSpatialGDKEditorToolbarModule::OnPropertyChanged(UObject* ObjectBeingModif
 	}
 }
 
+void FSpatialGDKEditorToolbarModule::CacheSpatialObjects(uint32 SpatialFlags)
+{
+	// Load the asset registry module
+	FAssetRegistryModule& AssetRegistryModule = FModuleManager::LoadModuleChecked<FAssetRegistryModule>(FName("AssetRegistry"));
+	IAssetRegistry& AssetRegistry = AssetRegistryModule.Get();
+
+	// Before running the interop generator, ensure all blueprint classes that have been tagged with 'spatial' are loaded
+	TArray<FAssetData> AssetData;
+	uint32 SpatialClassFlags = 0;
+	AssetRegistry.GetAssetsByClass(UBlueprint::StaticClass()->GetFName(), AssetData, true);
+	for (auto& It : AssetData)
+	{
+		if (It.GetTagValue("SpatialClassFlags", SpatialClassFlags))
+		{
+			if (SpatialClassFlags & SpatialFlags)
+			{
+				FString ObjectPath = It.ObjectPath.ToString() + TEXT("_C");
+				UClass* LoadedClass = LoadObject<UClass>(nullptr, *ObjectPath, nullptr, LOAD_EditorOnly, nullptr);
+				checkf(LoadedClass, TEXT("Failed to load blueprint class %s"), *ObjectPath);
```

Comment:
I think asserting is a bit rough here, let's treat this the same as a reliable multicast RPC error. Asserting in editor can lead to loss of data.

Reply:
Q about loss of data: do you mean that because this will cause an unresolvable breakpoint, VS will exit without saving user changes when user run codegen?

How does this differ from an assertion at runtime from unsaved blueprints in editor?